include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -461,39 +330,202 @@ class union_composition
+
+        constexpr size_t fixed_size = max_of_alphabet_sizes_v<alphabet_types...>;
+        using padded_value_to_char_t = std::array<char_type, fixed_size>;
+        using value_to_char_tables_t = std::array<padded_value_to_char_t, table_size>;
+
+        // store all `value_to_char` tables of each alphabet in one std::array
+        constexpr std::array value_to_char_tables = value_to_char_tables_t
+        {
+            make_value_to_char_table<fixed_size>(alphabet_types{})...
+        };
+
+        using value_to_char_t = std::array<char_type, table_size>;
+        value_to_char_t value_to_char{};
+        for (size_t i = 0u, value = 0u; i < table_size; ++i)
+            for (size_t k = 0u; k < value_sizes[i]; ++k, ++value)
+                value_to_char[value] = value_to_char_tables[i][k];
+
+        return value_to_char;
+    }
+
+    /*!\brief Returns an map at compile time where the key is the char of one of the
+     * alphabets and the value is the corresponding rank over all alphabets (by
+     * conflict will default to the first).
+     *
+     * ```cpp
+     * constexpr std::array char_to_value = char_to_value_table<char, dna4, gap, dna5>();
+     * assert(char_to_value.size() == 256);
+     * assert(char_to_value['A'] == 0);
+     * assert(char_to_value['C'] == 1);
+     * assert(char_to_value['G'] == 2);
+     * assert(char_to_value['T'] == 3);
+     * assert(char_to_value['-'] == 4);
+     * assert(char_to_value['A'] == 0);
+     * assert(char_to_value['C'] == 1);
+     * assert(char_to_value['G'] == 2);
+     * assert(char_to_value['T'] == 3);
+     * assert(char_to_value['N'] == 9);
+     * assert(char_to_value['*'] == 0); // every other character defaults to 0
+     * ```
+     */
+    static constexpr auto char_to_value_table() noexcept
+    {
+        constexpr size_t value_size = sum_of_alphabet_sizes_v<alphabet_types...>;
+        using rank_t = detail::min_viable_uint_t<value_size>;
+
+        constexpr size_t table_size = 1 << (sizeof(char_type) * 8);
+        constexpr std::array value_to_char = value_to_char_table();
+
+        using array_t = std::array<rank_t, table_size>;
+
+        array_t char_to_value{};
+        for (size_t i = 0u; i < value_to_char.size(); ++i)
+        {
+            rank_t & old_entry = char_to_value[value_to_char[i]];
+            bool is_new_entry = value_to_char[0] != value_to_char[i] && old_entry == 0;
+            if (is_new_entry)
+                old_entry = static_cast<rank_t>(i);
+        }
+        return char_to_value;
+    }
+
+    //!\brief Compile-time generated lookup table which contains the prefix sum up to the position of each alphabet.
+    //!\sa partial_sum_of_alphabet_sizes
+    static constexpr std::array partial_sum_sizes = partial_sum_of_alphabet_sizes<alphabet_types...>();
+
+    //!\brief Compile-time generated lookup table which maps the rank to char.
+    //!\sa value_to_char_table
+    static constexpr std::array value_to_char = value_to_char_table();
```

Comment:
A hunder lines later this is the actual table that is relevant for everything else in this file. Strangely it has the least amount of documentation and it isn't called table, although it's the only thing where it might make sense to call it table, because it is actually data structure.